sprite.py

```diff
@@ -60,3 +67,29 @@ def has_position(self, x, y, t):
         x and y do not have to be valid Grid coordinates.
         """
         return And(self.x(t) == x, self.y(t) == y)
+
+def draw_sprite(ctx:display.CellContext, sprite, t_start, t_end, sprite_fn):
+    """
+    If sprite is in the given cell at any time t between t_start (inclusive) and t_end (exclusive), rotates the drawing
+    context so that the sprite is pointing in the positive X direction at time t, and calls sprite_fn(ctx, t)
+    """
+    for t in range(t_start, t_end):
+        if ctx.model.eval(sprite.in_cell(ctx.cell, t)):
+            dir = ctx.model.eval(sprite.dir(t))
+            vx, vy = dir_to_vector[dir]
+            with display.transform_drawing_context(ctx, display.rotation_matrix_for_vector(vx, vy)):
+                sprite_fn(ctx, t)
+
+def make_frames(grid, frame_breaks, frames_per_row, spacing=3):
```

Comment:
The new `Display.display_all_grids()` is almost exactly what you just asked for. It doesn't care about what you want each grid to mean; it's going to display them where you said to put them, and hand you whatever extra data you said to attach to each one. I wrote it to be slightly more general because I was planning to use it to display e.g. the givens in `skyscrapers`, before I realized they weren't actually stored in grids. This `make_frames()` is just a helper to build the augmented grids `display_all_grids()` wants, in a fairly common sprite-based case. Are you asking for a slightly more flexible helper in between the regimented "n per row" of `make_frames()` and the freeform "put grids wherever you want" of `display_all_grids()`?

Reply:
Also, if it doesn't need to know the end of the last frame here (because it's just going to have indices anyway instead of `extra` as it does now), this doesn't need to add an extra value to `frame_breaks`.

Reply:
I get a lot happier once this all uses indices instead of `extra` and `frame_breaks`; my biggest beef here was with that extra value. After that change, I would probably factor this slightly differently myself; I'm not sure why `display_all_grids` needs to be quite so general in terms of getting (x, y) coordinates for each grid, so I would probably push the spacing and coordinate computation out of `make_frames` and into `display_all_grids`, and then all that's left of `make_frames` is `chunked`, which I figure users can probably call themselves. But I wouldn't be upset if that isn't how you end up slicing it.